### PR TITLE
Secrets: do not throw panics at startup

### PR DESCRIPTION
**Pull request by techmanmike**
Purpose: Don't require env-vars at startup and don't panic

Changes:
- Removed isTestMode
- Removed clients
- Renamed SECRETS.md to README.md

### CODE REVIEW

File: secrets/README.md [hunk 1]
```diff
@@ -32,8 +30,31 @@ func BasicExamples() {
 	fmt.Printf("The answer to the secret is '%s' (err='%v')\n", answer, err)
 
 	// or if you need secrets from another region other than the one you are running in use
-	sm := secrets.NewAWSSecretsManager(ctx, "a-different-region")
+	sm, err := secrets.NewAWSSecretsManager(ctx, "a-different-region")
 	answer, err = sm.Get(ctx, "my-test-secret2")
 	fmt.Printf("The answer to the secret2 is '%s' (err='%v')\n", answer, err)
+
+	// of if you want to have a custom client that
```
Comment (kimrogers) [inline secrets/README.md:37]: Is this a half finished thought?

File: secrets/package.go [hunk 2]
```diff
@@ -2,54 +2,34 @@ package secrets
...
+// Secrets can be mocked by clients for testing purposes.
+type Secrets interface {
+	Get(ctx context.Context, secretKey string) (string, error)
 }
 
+// DefaultAWSSecretsManager is a public *AWSSecretsManager used for package level methods.
+var DefaultAWSSecretsManager Secrets = nil
+
 // Get retrives the secret from AWS SecretsManager.
 func Get(ctx context.Context, secretKey string) (string, error) {
```
Comment (davidlopezre) [inline secrets/package.go:17]: Since these methods already return an error, I reckon it is better to return the error here than panicking.

Comment (techmanmike) [reply secrets/package.go:17]: Agree - I've fixed this in the latest commit.

File: secrets/README.md (hunk 1 above)
Comment (techmanmike) [reply secrets/README.md]: Fixed!

**Review by kimrogers — APPROVED** [review]
Thanks Mike, looks good.

**Pull request merged**